packages/dcos-launch/extra/launch/config.py

```diff
@@ -49,375 +21,256 @@ def expand_path(path: str, relative_dir: str) -> str:
-    ],
-    'default': {
-        'ssh_port': '22',
-        'ssh_private_key': calculate_ssh_private_key,
-        'instance_ami': calculate_instance_ami,
-        'os_name': 'cent-os-7-prereqs',
-        'key_helper': 'false',
-        'zen_helper': 'false'
-    },
-    'conditional': {
-        'provider': {
-            'aws': {
-                'default': {
-                    # allow untest-able deployment
-                    'ssh_user': '',
-                    'ssh_private_key_filename': ''
-                },
-                'must': {
-                    # TODO DCOS-14048: [gen.internals] allow user providing arguments
-                    # for a 'must' if the arguments agree
-                    'platform': 'aws'
-                }
-            },
-            'azure': {
-                'default': {
-                    # allow untest-able deployment
-                    'ssh_user': '',
-                    'ssh_private_key_filename': ''
-                },
-                'must': {
-                    # TODO DCOS-14048: [gen.internals] allow user providing arguments
-                    # for a 'must' if the arguments agree
-                    'platform': 'azure'
-                }
-            },
-            'onprem': {
-                'default': {
-                    'prevalidate_onprem_config': 'false',
-                    'num_public_agents': '0',
-                    'num_private_agents': '0',
-                    'installer_port': '9000',
-                    'admin_location': '0.0.0.0/0'
-                },
-                'must': {
-                    'onprem_dcos_config_contents': calculate_dcos_config_contents,
-                    'ssh_user': calculate_ssh_user,
-                    'cluster_size': calculate_cluster_size
-                },
-            },
-        },
-        'key_helper': {
-            'true': {
-                'must': {
-                    # key input not applicable if helper is used
-                    'aws_key_name': 'unset',
-                    'ssh_private_key_filename': 'unset',
-                    'ssh_private_key': 'unset'}},
-            'false': {}
-        }
-    }
-}
+    # use the intermediate provider-validated config to add the platform schema
+    platform = basic_validator.normalized(config)['platform']
+    if platform == 'aws':
+        basic_validator.schema.update(AWS_PLATFORM_SCHEMA)
+        if provider == 'onprem':
+            basic_validator.schema.update(AWS_ONPREM_SCHEMA)
+    elif platform == 'azure':
+        basic_validator.schema.update(AZURE_PLATFORM_SCHEMA)
+    else:
+        raise NotImplementedError()
+
+    # create a strict validator with our final schema and process it
+    final_validator = LaunchValidator(basic_validator.schema, config_dir=config_dir, allow_unknown=False)
+    if not final_validator.validate(config):
+        raise_errors(final_validator)
+    return final_validator.normalized(config)
+
+
+COMMON_SCHEMA = {
```

Comment:
Would it be beneficial to pull the `_SCHEMA` dictionaries to another file? This would provide separation between functionality (`config.py`) and rules (schema definitions).